[EMCAL-630] Skip TRU and LEDMON cells in raw->cell conversion

mfasDa opened the pull request:
TRU and LEDMON need to be discarded in the
raw to cell conversion as they have different geometry
mapping than FEC cells and then raw fitting procedure
cannot be applied to them.

mfasDa committed:
[EMCAL-630] Skip TRU and LEDMON cells in raw->cell conversion

TRU and LEDMON need to be discarded in the
raw to cell conversion as they have different geometry
mapping than FEC cells and then raw fitting procedure
cannot be applied to them.

mfasDa commented:
TRU decoding is currently be worked on. LEDMON cells are only used in the LED DA, here we need to discuss further what to do with them. For the ongoing tests we need the changes from this commit as several SMs have the LEDMON included.

shahor02 commented on Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx:
```diff
@@ -123,6 +123,7 @@ void RawToCellConverterSpec::run(framework::ProcessingContext& ctx)
       auto found = cellBuffer.find(currentIR);
       if (found == cellBuffer.end()) {
         currentCellContainer = std::make_shared<std::vector<Cell>>();
+        std::cout << "Setup cell container with " << currentCellContainer->size() << " entries" << std::endl;
```
please use LOG instead of the std::cout

mfasDa replied:
Ooops, sorry, these are leftover from debugging, will remove them again ...

shahor02 commented on Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx:
```diff
@@ -244,6 +305,7 @@ void RawToCellConverterSpec::run(framework::ProcessingContext& ctx)
   for (auto [bc, cells] : cellBuffer) {
     mOutputTriggerRecords.emplace_back(bc, triggerBuffer[bc], mOutputCells.size(), cells->size());
     if (cells->size()) {
+      std::cout << "Event has " << cells->size() << " cells" << std::endl;
```
idem

mfasDa replied:
See above

shahor02 reviewed (changes requested):
@mfasDa see cosmetic comments below. 
What is the reason to check for the invalid channels (where they come from)?

mfasDa commented:
@shahor02 The SRU payload contains 4 channel types

- High gain
- Low gain
- TRU (L0 trigger)
- LEDMON

The raw fitter can only work handle FEC cells (high- and low-gain). For L0 trigger a separate reconstruction is being worked on. The problem is however the LEDMON cells (dedicated sensors), which are only used in the LED DA. The mapping in the geometry only works for FEC cells, but we recently had them enabled and they lead to invalid cell IDs leading to a crash when processing. They need to be removed from the raw fitter. How we propagate then to the LED DA we will figure out later.

TRU cells we will reintroduce later once the porting is done, in a separate cell container (will be only used for the trigger QC).

mfasDa commented:
@shahor02 The couts are removed (second one changed to LOG(DEBUG). Could you please merge once the CI tests pass?

mfasDa committed:
[EMCAL-630] Decrease verbosity of RCU trailer

Printing of RCU trailer should be enabled explicitly,
in addition to high debug level. It should stay in the
code for debug purpose, but is not enabled by
by default

shahor02 commented:
Merging although the fullCI crashed in the gpu-reconstruction. Could not reproduce the crash locally but it should be unrelated to this PR anyway.
